### PR TITLE
[BottomNavigation] Fix image updates in ItemView

### LINKED ISSUE #6846: [BottomNavigation] Update the itemView's selected state after it is updated
Changing an unselected UITabBarItem's image would present in its selected color.

---
#### Internal data
- Associated internal bug: [b/128432654](http://b/128432654)

### PR DESCRIPTION
**Pull request by romoore**
When the `selectedImage` or `image` values were changed on
MDCBottomNavigationItemView, it was accidentally updating the view's
UIImageView regardless of its current selection state. By checking the
current state and only updating as-needed, changes to the icons don't
show up incorrectly.

Fixes #6846

### CODE REVIEW

File: components/BottomNavigation/tests/snapshot/MDCBottomNavigationItemViewSnapshotTests.m [hunk 1]
```diff
@@ -316,7 +316,7 @@ - (void)testChangeIconWhenNotSelected {
   [self generateAndVerifySnapshot];
 }
 
-- (void)testChangeIconWhenSelected {
+- (void)testChangeUnselectedImageWhenSelected {
```
Comment (romoore) [inline components/BottomNavigation/tests/snapshot/MDCBottomNavigationItemViewSnapshotTests.m]: This should set the selectedIcon as a Given

Comment (romoore) [reply components/BottomNavigation/tests/snapshot/MDCBottomNavigationItemViewSnapshotTests.m]: Also method above.

**Comment by romoore** [comment]
Please add unit tests.

**Comment by romoore** [comment]
Ready for review.

Should not be merged before https://github.com/material-components/material-components-ios/pull/6855 lands and the changes merged into this PR.

**Review by codeman7 — APPROVED** [review]

**Pull request merged**